### PR TITLE
Implements Cull uncommon card Channeling

### PR DESCRIPTION
**Pull request by Cartopol**
1: Play the top card of your deck for each incoming attack. Exhaust.

- Uses ReflectionHacks to get access to private properties of AbstractMonster. Not sure if this is desired.

### CODE REVIEW

File: src/main/java/stsjorbsmod/cards/cull/Channeling.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    @Override
+    public void use(AbstractPlayer p, AbstractMonster abstractMonster) {
+        for (int i = 0; i < magicNumber; ++i) {
+            this.addToBot(new PlayTopCardAction(AbstractDungeon.getCurrRoom().monsters.getRandomMonster(null, true, AbstractDungeon.cardRandomRng), false));
+        }
+    }
+
+    @Override
+    public String getRawDynamicDescriptionSuffix() {
+        return magicNumber == 1 ? EXTENDED_DESCRIPTION[0] : EXTENDED_DESCRIPTION[1];
```
Comment (wang429) [inline src/main/java/stsjorbsmod/cards/cull/Channeling.java]: For localization purposes, this assumes that other languages don't have special pluralization forms for 0 and 2, which is false. I'm not sure we've done the best job with localization language compatibility anyways. Might not be totally important right now.

Comment (dbjorge) [reply src/main/java/stsjorbsmod/cards/cull/Channeling.java]: Sadly, neither we nor the base game do a usable job of handling languages with non-english-like pluralization forms

Comment (dbjorge) [inline src/main/java/stsjorbsmod/cards/cull/Channeling.java]: The card description probably shouldn't reveal this much information if the player has runic dome; maybe have this check for `IntentUtils.areIntentsHidden()` and return empty string in that case

File: src/main/java/stsjorbsmod/cards/cull/Channeling.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    private static final int COST = 1;
+
+    public Channeling() {
+        super(ID, COST, TYPE, COLOR, RARITY, TARGET);
+        this.exhaust = true;
+        baseMagicNumber = 0;
+    }
+
+    private boolean isMonsterAttacking(AbstractMonster m) {
+        return m.intent.equals(AbstractMonster.Intent.ATTACK) ||
```
Comment (dbjorge) [inline src/main/java/stsjorbsmod/cards/cull/Channeling.java]: Can replace this with `IntentUtils.isAttackIntent(m.intent)`

File: src/main/java/stsjorbsmod/cards/cull/Channeling.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import basemod.ReflectionHacks;
+import com.megacrit.cardcrawl.actions.common.PlayTopCardAction;
+import com.megacrit.cardcrawl.characters.AbstractPlayer;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+import stsjorbsmod.JorbsMod;
+import stsjorbsmod.cards.CustomJorbsModCard;
+import stsjorbsmod.characters.Cull;
+
+public class Channeling extends CustomJorbsModCard {
```
Comment (dbjorge) [inline src/main/java/stsjorbsmod/cards/cull/Channeling.java]: Would be nice to add

```java
@Override
public boolean shouldGlowGold() {
    return IntentUtils.playerCanSeeThatAnyEnemyIntentMatches(IntentUtils::isAttackIntent);
}
```

File: src/main/java/stsjorbsmod/cards/cull/Channeling.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    @Override
+    public int calculateBonusMagicNumber() {
+        int channelCount = 0;
+        for (AbstractMonster m : AbstractDungeon.getCurrRoom().monsters.monsters) {
+            if (!m.isDying && m.currentHealth > 0 && !m.isEscaping) {
+                if (isMonsterAttacking(m)) {
+                    if ((boolean)ReflectionHacks.getPrivate(m, AbstractMonster.class, "isMultiDmg")) {
+                        channelCount += (int)ReflectionHacks.getPrivate(m, AbstractMonster.class, "intentMultiAmt");
```
Comment (dbjorge) [inline src/main/java/stsjorbsmod/cards/cull/Channeling.java]: This is a nice touch, I hadn't considered how this should interact with multiattacks when I initially read the card description but I think this is a good interaction.

**Pull request merged**